<launch>
  <param name="use_sim_time" value="true" />
  <node name="RVIZ" pkg="rviz" type="rviz" args=" -d $(find sonar_vis)/param/ddk_tof_repro.rviz" />

  <arg name="mav_name" default="dragonfly26"/>
  <arg name="cam_name" default="tof"/>
  <arg name="map_frame_id" default="$(arg mav_name)/odom"/>
  <arg name="base_frame_id" default="royale_royale_camera_link"/>
  <arg name="points_topic" default="/$(arg mav_name)/tof/reproject"/>
  <arg name="points_topic2" default="/$(arg mav_name)/tof/reproject_sonar"/>
  <arg name="sensor_range" default="5.0" />

  <group ns="$(arg mav_name)">
    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server">

      <!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
      <param name="frame_id" type="string" value="$(arg map_frame_id)" />

      <param name="resolution" type="double" value="0.05" />

      <param name="base_frame_id" type="string" value="$(arg base_frame_id)" />

      <!-- maximum range to integrate (speedup!) -->
      <param name="sensor_model/max_range" value="$(arg sensor_range)" />

      <param name="filter_ground" type="bool" value="false"/>
      <param name="ground_filter/distance" value="0.1"/>
      <param name="ground_filter/plane_distance" value="0.07"/>

      <!-- -0.5 to 2.5-->
      <param name="occupancy_min_z" value="-0.5"/>
      <param name="occupancy_max_z" value="2.5"/>
      <param name="pointcloud_min_z" value="-0.5"/>
      <param name="pointcloud_max_z" value="2.5"/>

      <!-- data source to integrate (PointCloud2) -->
      <remap from="cloud_in" to="$(arg points_topic)" />
    </node>
  </group>

  <group ns="$(arg mav_name)_sonar">
    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server2">

      <!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
      <param name="frame_id" type="string" value="$(arg map_frame_id)" />

      <param name="resolution" type="double" value="0.05" />

      <param name="base_frame_id" type="string" value="$(arg base_frame_id)" />

      <!-- maximum range to integrate (speedup!) -->
      <param name="sensor_model/max_range" value="$(arg sensor_range)" />

      <param name="filter_ground" type="bool" value="false"/>
      <param name="ground_filter/distance" value="0.1"/>
      <param name="ground_filter/plane_distance" value="0.07"/>

      <!-- -0.5 to 2.5-->
      <param name="occupancy_min_z" value="-0.5"/>
      <param name="occupancy_max_z" value="2.5"/>
      <param name="pointcloud_min_z" value="-0.5"/>
      <param name="pointcloud_max_z" value="2.5"/>

      <!-- data source to integrate (PointCloud2) -->
      <remap from="cloud_in" to="$(arg points_topic2)" />
    </node>
  </group>

  <arg name="sonar_frame_id" default="sonar"/>
  <node pkg="tf2_ros" type="static_transform_publisher" name="sonar_static_publisher" args="0.05 0.0 0.05 0 0 0 $(arg base_frame_id) $(arg sonar_frame_id)" />
  <node pkg="sonar_vis" type="maxsonar_listener2.py" name="pc2_gen" output="screen">
    <param name="sensor_range" value="$(arg sensor_range)" />
  </node>

</launch>